document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/exemptions/ApiToExemptionsMapper.java

```diff
@@ -1,13 +1,75 @@
 package uk.gov.companieshouse.document.generator.company.report.mapping.mappers.exemptions;
 
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
 import org.springframework.web.context.annotation.RequestScope;
+import uk.gov.companieshouse.api.model.exemptions.ExemptionItemsApi;
 import uk.gov.companieshouse.api.model.exemptions.ExemptionsApi;
 import uk.gov.companieshouse.document.generator.company.report.mapping.model.document.items.exemptions.Exemptions;
 
+import java.util.List;
+
 @RequestScope
 @Mapper(componentModel = "spring", uses = {ApiToExemption.class})
-public interface ApiToExemptionsMapper {
+public abstract class  ApiToExemptionsMapper {
+
+    public abstract Exemptions apiToExemptionsMapper(ExemptionsApi exemptionsApi);
+
+    @AfterMapping
+    void checkActiveExemption(ExemptionsApi exemptionsApi, @MappingTarget Exemptions exemptions) {
+
+        List<ExemptionItemsApi> disclosureTransparencyRulesChapterFiveAppliesList;
+        List<ExemptionItemsApi> pscExemptAsSharesAdmittedOnMarketList;
+        List<ExemptionItemsApi> pscExemptAsTradingOnRegulatedMarketList;
+        List<ExemptionItemsApi> pscExemptAsTradingOnUkRegulatedMarketList;
+
+        if(exemptionsApi.getDisclosureTransparencyRulesChapterFiveApplies() != null) {
+             disclosureTransparencyRulesChapterFiveAppliesList = exemptionsApi.getDisclosureTransparencyRulesChapterFiveApplies().getItems();
+
+            exemptions.setActiveExemption(false);
+
+            for (ExemptionItemsApi exemptionItemsApi : disclosureTransparencyRulesChapterFiveAppliesList) {
+                if(exemptionItemsApi.getExemptTo() == null) {
+                    exemptions.setActiveExemption(true);
+                }
+            }
```

Comment:
We can write the for loop once and put into a private method to cut down on duplication and increase readability

`    private void checkForActiveExemptions(List<ExemptionItemsApi> exemptionTypeList, Exemptions exemptions) {
        for (ExemptionItemsApi exemptionItemsApi : exemptionTypeList) {
            if (exemptionItemsApi.getExemptTo() == null) {
                exemptions.setActiveExemption(true);
            }
        }
    }`

We can then call this by:

`checkForActiveExemptions(disclosureTransparencyRulesChapterFiveAppliesList, exemptions);`